C/cp/LuhnsAlgo.c

```diff
@@ -0,0 +1,81 @@
+/******************************************************************************
+
+Author - @Suvraneel 
+        Suvraneel Bhuin
+
+*Luhn's Algorithm for credit card validation*
+
+A credit card number consists of 16 decimal digits. The first (leftmost) digit corresponds to the type of the
+card (4 for VISA cards, 5 for MASTERCARDs, 3 for Diners Club and American Express cards, and so on).
+The next 6 digits correspond to the number of the bank that issues the card. The next 8 digits constitute the
+account number of the card-holder. The last (rightmost) digit is a check digit that determines whether the
+credit card number is valid or not. In order to determine the check digit, a checksum is computed as follows.
+We first add the digits at the even positions (that is, the second, fourth, sixth, ... , sixteenth digits) of the card
+number. Next, each digit in an odd position is doubled. If the doubled value exceeds 9, then the two digits
+are added. Finally, these doubled digits (after adjustment of 2-digit values) are added to the checksum. The
+check digit is so chosen that the checksum is a multiple of 10. Here is an example.
+Card Number   5        9   8         7   6         5   4   0   2   4   6         8   1   3   5         5
+Step 1        10       9   16        7   12        5   8   0   4   4   12        8   2   3   10        5
+Step 2        1+0 = 1  9   1+6 = 7   7   1+2 = 3   5   8   0   4   4   1+2 = 3   8   2   3   1+0 = 1   5
+
+Checksum = 1 + 9 + 7 + 7 + 3 + 5 + 8 + 0 + 4 + 4 + 3 + 8 + 2 + 3 + 1 + 5 = 70
+
+Since the checksum is divisible by ten, 5987654024681355 is a valid credit card number. Write a program
+that, upon input a 16-digit credit card number, determines whether the card number is valid. A 16-digit
+integer cannot be accommodated in a long int variable, so read the input as a string (a character array).
+*******************************************************************************/
+#include <stdio.h>
+#include <string.h>                              //header for string operations
+
+int main()
+{
+    char card[16];                               //character array
+    printf("Enter the credit card number:\n");
+    gets (card);                                 //reads a line from stdin and stores it into the 'card' character array
+    int checksum = 0;                            //initiates checksum to zero
+    for (int i=0; i<strlen(card) ; i++)          //iterates as many times as there are characters in the input string
+    {
+        int digit = card[i] - '0';
+        if (i%2==1){                             //even positions
+        checksum += digit;                       //digits added to checksum as it is
+        printf(" + %d", digit);
```

Comment:
Put the Comments above the Line of Code

Reply:
The inline codes right? 
Sure... right away.. Thanks